packages/block-library/src/page-list/use-convert-to-navigation-links.js

```diff
@@ -30,11 +37,57 @@ export function convertToNavigationLinks( pages = [] ) {
 				// Use a placeholder if the child appears before parent in list.
 				linkMap[ parent ] = { innerBlocks: [] };
 			}
+			// Although these variables are not referenced, they are needed to store the innerBlocks in memory.
 			const parentLinkInnerBlocks = linkMap[ parent ].innerBlocks;
 			parentLinkInnerBlocks.push( linkMap[ id ] );
 		}
 	} );
 
+	return navigationLinks;
+}
+
+/**
+ * Finds a navigation link block by id, recursively.
+ * It might be possible to make this a more generic helper function.
+ *
+ * @param {Array}  navigationLinks An array of navigation link blocks.
+ * @param {number} id              The id of the navigation link to find.
+ *
+ * @return {Object} The navigation link block with the given id.
+ */
+function findById( navigationLinks, id ) {
+	for ( const navigationLink of navigationLinks ) {
+		// Is this the link we're looking for?
+		if ( navigationLink.attributes.id === id ) {
+			return navigationLink;
+		}
+
+		// If not does it have innerBlocks?
+		if ( navigationLink.innerBlocks && navigationLink.innerBlocks.length ) {
+			const foundNavigationLink = findById(
+				navigationLink.innerBlocks,
+				id
+			);
+
+			// Does the innerBlocks have the link we're looking for?
+			if ( foundNavigationLink ) {
+				return foundNavigationLink;
+			}
+		}
+	}
+}
+
+export function convertToNavigationLinks( pages = [], parentPageID = null ) {
+	let navigationLinks = createNavigationLinks( pages );
+
+	// If a parent page ID is provided, only return the children of that page.
+	if ( parentPageID ) {
```

Comment:
Nit: I wonder if we should not instead filter `pages` and use `createNavigationLinks` on the filtered pages?

Reply:
I thought about that, but it's technically more complex as `pages` isn't a nested array, so we'd have to filter on a parent id, and then all the parents of that id, and then all the parents of that one etc. I think this approach is simpler.